changelog/0076-upgrade-to-es-5.yml

```diff
@@ -0,0 +1,279 @@
+title: Elasticsearch upgrade from 2.4.6 to 5.6.16
+key: upgrade-to-es-5
+date: 2024-01-14
+optional_per_env: no
+min_commcare_version: 3208896a555c69be119bbd7f0d06de90120b6d14
+max_commcare_version:
+context: |
+  Upgrade to Elasticsearch 5.
+
+  CommCare HQ releases after Mar 1, 2024 will not support Elasticsearch 2.x.
+  So we strongly recommend applying this change before then.
+
+details: |
+  As part of our ongoing effort to keep CommCare HQ up to date with the latest tools and
+  libraries we have upgraded Elasticsearch from version 2.4.6 to 5.6.16.
+
+update_steps: |
+  This process should be followed after Reindexing is complete for all indices. Refer to the previous [changelog](https://commcare-cloud.readthedocs.io/en/latest/changelog/0075-reindex-all-indexes-for-es-upgrade.html) for further details.
+
+  After you ensure that you have followed the steps mentioned in previous changelog, you are all set to move with upgrading your Elasticsearch cluster to version 5.
+
+  ### Pre-Upgrade Steps
+
+  1. Start by ssh-ing into the control machine.
+
+    ```
+      cchq <env> ssh control
+    ```
+
+  2. Start a new tmux session
+
+    ```
+      tmux new -s es_upgrade
+    ```
+
+  3. For commands including `${ENV}` and `${ES_HOST}` below to work, make sure to set `ENV` to the name of your environment, and `ES_HOST` to the IP address of one of your ES nodes:
+    
+    ```
+      ENV=<env>
+      ES_HOST=$(cchq ${ENV} lookup elasticsearch:0 | cut -d' ' -f1)
+    ```
+
+  4. Delete old and unused CommCare Indices by running the following commands. The command will ask you confirm the deletion by typing in the index cname. 
+    ```
+      cchq ${ENV} django-manage elastic_sync_multiplexed delete apps
```

Comment:
Does this command include a safety check that prevents deletion if it would delete the primary index?

Reply:
Yep. It is true. The command will check the settings and will only work if multiplexer is turned off and indices are swapped. And then will an attempt to delete older index that is no longer used.